exonum/src/blockchain/builder.rs

```diff
@@ -177,29 +121,48 @@ impl InstanceCollection {
             id,
             name: name.into(),
         };
-        let instance_config = InstanceConfig::new(spec, None, params.into_bytes());
+        let instance_config = ConfiguredInstanceSpec {
+            instance_spec: spec,
+            constructor: params.into_bytes(),
+        };
         self.instances.push(instance_config);
         self
     }
 }
 
+impl From<InstanceCollection> for (Box<dyn ServiceFactory>, InstanceConfig) {
```

Comment:
Seems like `InstanceCollection` now exists only to collect factories and instances together, and then split them back. Do we need it at all?

Reply:
I also noticed that, but didn't end up with decision if its worth refactoring within this PR.

Reply:
Anyway, I'd create corresponding task and write TODO comment in code in order not to forget about it.

Reply:
ECR-3913